### PR TITLE
Argument separation in CSV/flat JSON

### PR DESCRIPTION
**Pull request by AlGreed**
As i mentioned in the thread #618 , in my opinion it would be enough to have for this case a csv table.  Additionally JSON format for the user perspective is not enough representable to get an overview of comparison of encryption ciphers.

This scenario works only for testing of ciphers.
For example:
./testssl.sh -x CBC --csv host:port
./testssl.sh -x SSL_CK_RC2_128_CBC_EXPORT40_WITH_MD5  --csv host:port

So: 
cbc cyphers

**About code:**
Some fields were like "ECDH 570", other like " RSA(512)"

If a cipher was not available, so key exchange was printed colorless and in round brackets 
like "RSA(512)".
If a cipher was available, so bits in key exchange was printed separate and colorful according to cipher strength like "ECDH 570"

For now  i made it more consistent and bits are always printed in round brackets, but still colorful only for available ciphers.

image

### CODE REVIEW

File: testssl.sh [hunk 1]
```diff
@@ -760,8 +760,17 @@ fileout() {
      if ( "$do_pretty_json" && [[ "$1" == "service" ]] ) || show_finding "$severity"; then
          local finding=$(strip_lf "$(newline_to_spaces "$(strip_quote "$3")")")
          [[ -f "$JSONFILE" ]] && (fileout_json_finding "$1" "$severity" "$finding" "$cve" "$cwe" "$hint")
-         "$do_csv" && \
-              echo -e \""$1\"",\"$NODE/$NODEIP\",\"$PORT"\",\""$severity"\",\""$finding"\",\""$cve"\",\""$cwe"\",\""$hint"\"" >> "$CSVFILE"
+
+         if "$do_csv"; then
+            if "$do_test_just_one"; then
```
Comment (AlGreed) [inline testssl.sh]: a boolean variable "do_test_just_one" will be set to true only in the case of -x:

```
-x|-x=*|--single[-_]cipher|--single[-_]cipher=*)
     do_test_just_one=true
     single_cipher=$(parse_opt_equal_sign "$1" "$2")
     [[ $? -eq 0 ]] && shift
     ;;
```
 Is it a right name for it? How i understand if we type "-x CBC" -> all ciphers with cipher block chaining will be analysed. Maybe could we rename it to something like "only_specific_ciphers"? Otherwise "if "$do_test_just_one"; then" is abit confusing in my opinion by reading of the code.

File: testssl.sh [hunk 2]
```diff
@@ -841,7 +850,11 @@ csv_header() {
           CSVHEADER=false
      else
           [[ -e "$CSVFILE" ]] && fatal "\"$CSVFILE\" exists. Either use \"--append\" or (re)move it" 1
-          echo "\"id\",\"fqdn/ip\",\"port\",\"severity\",\"finding\",\"cve\",\"cwe\",\"hint\"" > "$CSVFILE"
+          if "$do_test_just_one"; then
+             echo "\"ID\",\"FQDN/IP\",\"Port\",\"Severity\",\"Hexcode\",\"OpenSSL CSN\",\"Key Exchange\",\"Encryption\",\"Bits\",\"RFC CSN\",\"Availability\",\"CVE\",\"CWE\",\"Hint\"" > "$CSVFILE"
```
Comment (AlGreed) [inline testssl.sh]: Not sure that we need CWE, CVE and maybe Hint for this case? to remove from table?

File: testssl.sh (hunk 1 above)
Comment (drwetter) [inline testssl.sh]: yes, we could change this to ~ ``do_cipher_matched``. ``test_just_one()``. That anyway wasn't consistent and then would be needed to be renamed then to ``run_cipher_matched()``.

Comment (drwetter) [inline testssl.sh]: But lets keep this for a separate PR pls

File: testssl.sh (hunk 2 above)
Comment (drwetter) [inline testssl.sh]: For this case not, at least not yet. We would have to look at single ciphers in order to tell whether it represents something good or bad -- which we don't do.

File: testssl.sh [hunk 3]
```diff
@@ -2417,7 +2430,7 @@ neat_list(){
           fi
           return 0
      fi
-     read what_dh bits <<< "$kx"
+     read what_dh bits <<< "$(echo "$kx" | awk -F"(" '{a=$1" ("$2; print a}')"
```
Comment (drwetter) [inline testssl.sh]: Not sure about the reason of the change here but if there's one, one minor thing: I would rather avoid a pipe like  ``read what_dh bits <<< "$(awk -F"(" '{a=$1" ("$2; print a}') <<< "$kx" "

File: testssl.sh [hunk 4]
```diff
@@ -2624,7 +2637,7 @@ test_just_one(){
                     ciphers_found[i]=true
                     if [[ ${kx[i]} == "Kx=ECDH" ]] || [[ ${kx[i]} == "Kx=DH" ]] || [[ ${kx[i]} == "Kx=EDH" ]]; then
                          dhlen=$(read_dhbits_from_file "$TMPFILE" quiet)
-                         kx[i]="${kx[i]} $dhlen"
+                         kx[i]="${kx[i]}($dhlen)"
```
Comment (drwetter) [inline testssl.sh]: Nah, that looks odd for everything else than CSV output. Also the round brackets are colorized.

**Comment by drwetter** [comment]
@AlGreed: The part except in neat_list() is fine, thank you! Could you please rebase and then do a separate PR for this part? For CWE. CVE and hints just omit them
